modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/web/internal/product/navigation/control/menu/test/EditArticleHeaderProductNavigationControlMenuEntryTest.java

```diff
@@ -0,0 +1,250 @@
+
+import org.springframework.mock.web.MockHttpServletRequest;
+
+/**
+ * @author Mikel Lorza
+ */
+@RunWith(Arquillian.class)
+public class EditArticleHeaderProductNavigationControlMenuEntryTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_group = GroupTestUtil.addGroup();
+
+		LayoutTestUtil.addTypePortletLayout(_group);
+
+		ServiceContextThreadLocal.pushServiceContext(
+			ServiceContextTestUtil.getServiceContext(_group.getGroupId()));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderInAnotherView() throws Exception {
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			_group.getGroupId(),
+			JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(
+				_getMockHttpServletRequest(journalArticle)));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderWithGuestPermissions()
+		throws Exception {
+
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			_group.getGroupId(),
+			JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+		MockHttpServletRequest mockHttpServletRequest =
+			_getMockHttpServletRequest(journalArticle);
+
+		mockHttpServletRequest.addParameter(
+			"mvcRenderCommandName", "/journal/edit_article");
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderWithoutGuestPermissions()
+		throws Exception {
+
+		JournalArticle journalArticle = JournalTestUtil.addArticle(
+			_group.getGroupId(),
+			JournalFolderConstants.DEFAULT_PARENT_FOLDER_ID);
+
+		_resourcePermissionLocalService.deleteResourcePermissions(
+			_group.getCompanyId(), JournalArticle.class.getName(),
+			ResourceConstants.SCOPE_INDIVIDUAL,
+			journalArticle.getResourcePrimKey());
+
+		MockHttpServletRequest mockHttpServletRequest =
+			_getMockHttpServletRequest(journalArticle);
+
+		mockHttpServletRequest.addParameter(
+			"mvcRenderCommandName", "/journal/edit_article");
+
+		Assert.assertTrue(
+			_productNavigationControlMenuEntry.isShow(mockHttpServletRequest));
+	}
+
+	@Test
+	public void testIsShowJournalArticleHeaderWithoutJournalArticle()
+		throws Exception {
+
+		Assert.assertFalse(
+			_productNavigationControlMenuEntry.isShow(
+				_getMockHttpServletRequest()));
+	}
+
+	private MockHttpServletRequest _getMockHttpServletRequest()
+		throws Exception {
+
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequest(mockHttpServletRequest);
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			PortletServlet.PORTLET_SERVLET_REQUEST, mockHttpServletRequest);
+
+		MVCPortlet mvcPortlet = (MVCPortlet)_portlet;
```

Comment:
Q: Why are you casting this to `MVCPortlet`? It shouldn't be necessary.

Reply:
I thought it was necessary but I am seeing that no, so , I have deleted this casting.

thanks